api/python/provisioner/commands/sw_upgrade.py

```diff
@@ -122,6 +143,49 @@ def run(self, targets):  # noqa
         try:
             self._ensure_upgrade_repos_configuration()
 
+            local_minion = local_minion_id()
+
+            version_resolver = GetReleaseVersion()
+            release_info = version_resolver.run(targets=local_minion)  # str
+            release_info = json.loads(release_info)
+            cortx_version = (f"{release_info[ReleaseInfo.VERSION.value]}-"
+                             f"{release_info[ReleaseInfo.BUILD.value]}")
+
+            try:
+                yum_snapshots = self._get_pillar('upgrade/yum_snapshots',
+                                                 local_minion)
+            except ValueError:
+                yum_snapshots = {}  # assume yum_snapshots are empty
+
+            # TODO: we need to compare targets in yum_snapshots and targets
+            #  from parsed income parameter
+            #  for example:
+            #  yum_snapshots:
+            #    - <cortx_version>:
+            #       - srvnode-1: <id>
+            # and input targets = "*" for cluster with >= 2 nodes
+            # so possible scenario:
+            # provisioner sw_upgrade --targets="srvnode-1"
+            # provisioner sw_upgrade --targets="*"
+            if cortx_version not in yum_snapshots:
```

Comment:
I know that this conditional pillar update logic was initially requested but as it has been realized later it's not the best way since the state of yum might be shifted between two tries of sw upgrade. Could you remove that conditional so the pillar would be always updated with the current state? Thanks

Reply:
I agree. I updated the logic